Add trace function for matrix types

zhai-xiao opened the pull request:
Add trace function for matrix types

Signed-off-by: Xiao Zhai <zhaixiao43+git@gmail.com>

cary-ilm commented:
Thanks for the contribution.  Could you kindly add a test for trace() in src/ImathTest/testMatrix.cpp?

cary-ilm commented on src/Imath/ImathMatrixAlgo.h:
```diff
@@ -779,18 +779,17 @@ extractQuat (const Matrix44<T>& mat)
     Quat<T> quat;
 
     int nxt[3] = {1, 2, 0};
-    tr         = mat[0][0] + mat[1][1] + mat[2][2];
```
This is not correct. mat.trace() includes mat[3][3], but tr in the old formulation does not. Thus the test failure.

zhai-xiao replied:
Oh I see, sorry about that :( I'll revert

zhai-xiao commented:
> Thanks for the contribution. Could you kindly add a test for trace() in src/ImathTest/testMatrix.cpp?

Sure thing, tests added! Also I reverted the error I introduced in ImathMatrixAlgo.h. Thanks for reviewing @cary-ilm

cary-ilm reviewed (approved):
LGTM
